pkg/operator/cluster/controller.go

```diff
@@ -111,14 +111,19 @@ func (c *ClusterController) onAdd(obj interface{}) {
 
 	cluster.init(c.context)
 	if c.devicesInUse && cluster.Spec.Storage.AnyUseAllDevices() {
-		logger.Warningf("using all devices in more than one namespace not supported. ignoring devices in namespace %s", cluster.Namespace)
-		cluster.Spec.Storage.ClearUseAllDevices()
+		logger.Errorf("using all devices in more than one namespace not supported")
+		return
 	}
 
 	if cluster.Spec.Storage.AnyUseAllDevices() {
 		c.devicesInUse = true
 	}
 
+	if cluster.Spec.MonCount <= 0 {
+		logger.Errorf("mon count is 0 or less (given: %d), needs to be greater 0", cluster.Spec.MonCount)
```

Comment:
actually can we default to 3 if it is not specified? This way it is not a breaking change to the cluster.yaml